zenoh/src/sample.rs

```diff
@@ -326,6 +329,8 @@ pub struct Sample {
     pub kind: SampleKind,
     /// The [`Timestamp`] of this Sample.
     pub timestamp: Option<Timestamp>,
+    /// Quality of service settings this sample was sent with.
+    pub qos: QoS,
```

Comment:
I'm not a huge fan of how `Sample` keeps on getting more public fields. Maybe we can use the API break to make it a bit more private?